index.js

```diff
@@ -141,262 +95,224 @@ const filterData = (val, filterType, currentData) => {
     currentData = currentData.filter((book) => book.price <= val);
   }
   if (filterType === 'author') {
-    currentData = currentData.filter((book) =>
-      book.author.toLowerCase().includes(val.toLowerCase())
-    );
+    currentData = currentData.filter((book) => book.author.toLowerCase().includes(val.toLowerCase()));
   }
   if (filterType === 'publicationYear') {
     currentData = currentData.filter(
-      (book) => book.publicationYear.toString() === val
+      (book) => book.publicationYear.toString() === val,
     );
   }
   console.log('Filtered Examined Books :- ');
   console.log(currentData);
   return currentData;
 };
 
-//Here We Want to render Similar Books, firstly on the basis of the same genre and then on the basis of ascending order of price.
-//RULE: We want to display similar books with same genre and similarPrice Range in sorted Order, then the rest.
+// Here We Want to render Similar Books, firstly on the basis of the same genre and then on the basis of ascending order of price.
+// RULE: We want to display similar books with same genre and similarPrice Range in sorted Order, then the rest.
 const renderSimilarBooks = (currentBook) => {
```

Comment:
This method is too long and seems unoptimized, we can rewrite the method like below. Please verify
```
const renderSimilarBooks = (currentBook) => {
  const similarBooksMap = new Map();

  data.forEach((book) => {
    if (book.bookId !== currentBook.bookId) {
      const matchesGenre = book.genre.toLowerCase() === currentBook.genre.toLowerCase();
      const matchesPrice = book.price >= currentBook.price * 0.9 && book.price <= currentBook.price * 1.1;

      if (matchesGenre && matchesPrice) {
        similarBooksMap.set(book.bookId, book);
      } else if (matchesGenre || matchesPrice) {
        if (!similarBooksMap.has(book.bookId)) {
          similarBooksMap.set(book.bookId, book);
        }
      }
    }
  });

  const finalSimilarBookList = Array.from(similarBooksMap.values());

  console.log('All Similar Books :- ');
  console.log(finalSimilarBookList);
  
  similarTableBody.innerHTML = ''; // Clear old rows to avoid duplicates

  finalSimilarBookList.slice(0, 10).forEach((book) => {
    const coverImage = book.coverImage || defaultImage;
    const rowEntry = document.createElement('tr');
    rowEntry.innerHTML = `
      <td><img src="${coverImage}" alt="${book.bookName}" style="width:50px;height:50px; object-fit:contain;"></td>
      <td>${book.bookName}</td>
      <td>${book.genre}</td>
      <td>$ ${book.price}</td>
      <td>${book.author}</td>
      <td>${book.publicationYear}</td>`;
    similarTableBody.appendChild(rowEntry);
  });

  similarBookCount.innerHTML = `Showing <b>${Math.min(10, finalSimilarBookList.length)}</b> of <b>${finalSimilarBookList.length}</b> Similar Book(s)`;
  renderAllBooks();
};

```

Reply:
Done. I have made all the required changes but, the code you've given for rendering similar books seems inconsistent and it doesn't display the data in required order. So I have replaced the filtering logic with my code and the rendering logic with the code you've provided.